src/Microsoft.AspNet.Identity/RoleManager.cs

```diff
@@ -404,12 +404,12 @@ private IRoleClaimStore<TRole> GetClaimStore()
         /// <param name="user"></param>
         /// <param name="methodName"></param>
         /// <returns></returns>
-        protected async Task<IdentityResult> LogResultAsync(IdentityResult result,
+        protected Task<IdentityResult> LogResultAsync(IdentityResult result,
             TRole role, [System.Runtime.CompilerServices.CallerMemberName] string methodName = "")
         {
-            result.Log(Logger, Resources.FormatLoggingResultMessageForRole(methodName, await GetRoleIdAsync(role)));
+            result.Log(Logger, () => { return Resources.FormatLoggingResultMessageForRole(methodName, GetRoleIdAsync(role).Result); });
```

Comment:
This is called in the IdentityResult.Log method https://github.com/aspnet/Identity/blob/DeferedLogging/src/Microsoft.AspNet.Identity/IdentityResult.cs#L65 so it has to return some result and since the log methods are not async I do a block 

These are called only if the loglevel is enabled though